src/porepy/models/run_models.py

```diff
@@ -105,17 +105,15 @@ def time_step() -> bool:
         # step succeeded or failed.
         return solver.solve(model)
 
-    # Progressbars turned off or tqdm not installed:
-    if not params.get("progressbars", False) or not _IS_TQDM_AVAILABLE:
-        while not model.time_manager.final_time_reached():
-            time_step()
-
-    # Progressbars turned on:
-    else:
-        # Redirect the root logger, s.t. no logger interferes with with the
-        # progressbars.
-        with logging_redirect_tqdm([logging.root]):
-            # Time loop
+    # Redirect the root logger, s.t. no logger interferes with with the
+    # progressbars.
+    with logging_redirect_tqdm([logging.root]):
+        initial_time_step: float = model.time_manager.dt
```

Comment:
No need to `assert initial_time_step != 0` as below? See also comment regarding code placement.

Reply:
I am not sure whether that assertion is needed anyways. With an initial time step size
of 0, the simulation will never finish. My intuition would be that this should not be caught by
run_time_dependent_model, but by time_manager and raise
an Error. I removed the line below.